index.channel.ts

```diff
@@ -0,0 +1,441 @@
+        return this._buttonsFormat(envelope.message, options);
+      case OutgoingMessageFormat.carousel:
+        return this._carouselFormat(envelope.message, options);
+      case OutgoingMessageFormat.list:
+        return this._listFormat(envelope.message, options);
+      case OutgoingMessageFormat.quickReplies:
+        return this._quickRepliesFormat(envelope.message, options);
+      case OutgoingMessageFormat.text:
+        return this._textFormat(envelope.message, options);
+
+      default:
+        throw new Error('Unknown message format');
+    }
+  }
+
+  async sendMessage(
+    event: DiscordEventWrapper,
+    envelope: StdOutgoingEnvelope,
+    options: any,
+    context: any,
+  ): Promise<{ mid: string }> {
+    try {
+      this.logger.log('Discord Channel Handler: Sending message ...');
+      debugger;
+      const client = this.discordBotService.getClient();
+      const handler: DiscordChannelHandler = this;
+      const originalEvent:
+        | ButtonInteraction
+        | ChatInputCommandInteraction
+        | Message = event.getOriginalEvent();
+
+      const payload = await handler._formatMessage(envelope, options);
+      const message = event.getOriginalMessageContent();
+
+      if (originalEvent.channel.type === ChannelType.GuildText) {
+        // Create embed to show user's message
+        const embed = new EmbedBuilder()
+          .setColor(0x6cc853)
+          .addFields(
+            {
+              name: `${event.getSender().first_name}:`,
+              value: message,
+            },
+            {
+              name: 'Hexabot:',
+              value: payload?.content || '\u200B',
+            },
+          )
+          .setAuthor({
+            name: '\u200B',
+            iconURL: event.getProfile().avatarUrl,
+          });
+        if (payload.files) {
+          embed.setImage('attachment://' + payload.files[0].name);
+        }
+
+        const res = await (
+          originalEvent as ButtonInteraction | ChatInputCommandInteraction
+        ).followUp({
+          embeds: [embed],
+          components: payload.components,
+          files: payload.files,
+        });
+        return { mid: res.id };
+      }
+
+      if (originalEvent.channel.type === ChannelType.DM) {
+        const channel = await client.channels.fetch(
+          event.getRecipientForeignId(),
+        );
+        if (channel?.isTextBased()) {
+          const res: any = await (channel as TextChannel).send(payload);
+          return { mid: res.id };
+        }
+      }
+
+      // return { mid: '' };
+
+      // const channelId = event.getRecipientForeignId();
+      // const channel = await this.client.channels.fetch(channelId);
+
+      // if (channel?.isTextBased()) {
+      //   const res: any = await (channel as TextChannel).send(message);
+      //   return { mid: res.id };
+      // } else {
+      //   this.logger.log('Channel not found or is not text-based');
+      //   return { mid: '' };
+      // }
```

Comment:
Do we still need this ?

Reply:
We don't, with the current working implementation.